pycheribuild/projects/run_qemu.py

```diff
@@ -222,18 +225,18 @@ def add_smb_dir(directory, target, share_name=None, readonly=False):
         if self._hasPCI:
             qemuCommand += ["-device", "virtio-rng-pci"]
 
-        if self.config.wait_for_debugger:
-            self.info("QEMU is waiting for GDB to attach (using `target remote :1234`)."
+        if self.config.wait_for_debugger or self.config.debug_kernel:
+            gdb_port = randint(1200,1300) if self.config.gdb_random_port else 1234
+            self.info(f"QEMU is waiting for GDB to attach (using `target remote :{gdb_port}`)."
                       " Once connected enter 'continue\\n' to continue booting")
-
             def gdb_command(main_binary, breakpoint=None, extra_binary=None) -> str:
                 gdb_cmd = BuildGDB.getInstallDir(self, cross_target=CompilationTargets.NATIVE) / "bin/gdb"
                 # Set the sysroot to ensure that the .debug file is loaded from $SYSROOT/usr/lib/debug/boot/kernel
                 result = [gdb_cmd, main_binary, "--init-eval-command=set sysroot " + str(self.rootfs_path)]
                 # Once the file has been loaded set a breakpoint on panic() and connect to the remote host
                 if breakpoint:
                     result.append("--eval-command=break " + breakpoint)
-                result.append("--eval-command=target remote localhost:1234")
+                result.append(f"--eval-command=target remote localhost:{gdb_port}")
```

Comment:
Unfortunately cheribuild still needs to support python 3.5 so we can't use f-strings.

Reply:
With Ubuntu 20.04 released a week ago, we could start to retire 16.04 support and mandate at least 18.04. 16.04 only has another year left until you have to pay for extended maintenance (except for personal use), and I very much hope that we can kill off the remaining 16.04 Jenkins slaves imminently.

Reply:
Even after killing the jenkins slaves, we might still have external users of cheribuild that are using an older (but still supported Ubuntu version). Currently increasing the minimum to 3.6 really doesn't add much other than f-strings, so I don't see an urgent need to move to 3.6 minimum.

For other projects increasing the minimum supported release might be more useful.
I don't think we need to support 16.04 for every target supported by cheribuild.